### DISCUSSION TITLE
sysbuild board overlay

### DISCUSSION THREAD

**Discussion by jefflongo** [zephyrproject-rtos/zephyr#63067] [Q&A]
I want to compile MCUboot with my Zephyr application. According to [Zephyr docs](https://docs.zephyrproject.org/latest/build/sysbuild/index.html), sysbuild is a good way to achieve this. However, I'd like to specify some board overlay files when the sysbuild compiles MCUboot. By default, it will use an overlay from MCUboot.

From the documentation, I can provide an overlay file with `sysbuild/mcuboot.overlay`, but this does not solve the problem because my application supports multiple boards, and I'd like to use a different overlay for each.

A second solution from the documentation is to provide `sysbuild/mcuboot/boards/<BOARD>.overlay`, which does indeed allow me to have per-device overlays when compiling MCUboot, but introduces a new problem. In this approach, `APPLICATION_CONFIG_DIR` is set to `sysbuild/mcuboot` thus requiring me to provide a `prj.conf` (and the `prj.conf` from MCUboot is omitted). MCUboot's `prj.conf` provides some important config options that are necessary.

Neither of these approaches, from what I've tried so far, are satisfactory. As a workaround, I can pass `-Dmcuboot_DTC_OVERLAY_FILE=<path/to/overlay>` to west when building, but this gets tiresome when compiling for different boards. Is there a way to compile a sysbuild with a per-board devicetree overlay for a module while still inheriting the module's `prj.conf`?

**Comment by nordicjm** [comment]
Not really no, the idea is you copy the prj.conf file from that and store it so that you are not hit with unexpected changes if the project you are using suddenly changed their configuration (e.g. what if mcuboot decided to change the default key type from RSA to ECDSA?)

**Reply by jadejr** [reply]
just because you want an overlay doesn't mean you don't want the mcuboot defaults anymore.

**Reply by nordicjm** [reply]
Then you would do something like `-Dmcuboot_EXTRA_DTC_OVERLAY_FILE="/full/path/to/my.overlay"`

**Reply by achim-heyenzo** [reply]
Unfortunately, "/full/path" may depend on the OS, e.g. Windows "C:/ncs/v3.4.0/myapp" or Linux "/home/me/ncs/v3.4.0/myapp". 
I also tried a "../../relative/path", but that seems not to work.

It's annoying: 
I tried to put the "my.overlay" also in the "sysbuild/mcuboot/boards", but that fails as well without full-path.
